### PR TITLE
EOS-19490: Fix - hostname reset on reboot

### PR DESCRIPTION
**Pull request by SomwanshiAnjali**
hostname was getting reset to old one on reboot due to "operation not permitted" error
fix:
1. Remove `i` attribute from /etc/hostname 
2. Use hostnamectl command to set the hostname

Signed-off-by: Anjali Somwanshi <anjali.somwanshi@seagate.com>

### CODE REVIEW

File: srv/components/system/config/set_hostname.sls [hunk 1]
```diff
@@ -15,9 +15,12 @@
 # please email opensource@seagate.com or cortx-questions@seagate.com.
 #
 
+Update hostname file permissions:
+  cmd.run:
+    - name: "chattr -i /etc/hostname"
+
 Set hostname:
-  network.system:
```
Comment (83bhp) [inline srv/components/system/config/set_hostname.sls:19]: after changing the attr can we still use network.system way to update hostname? Or is there any other salt standard module to do the same?

**Review by 83bhp — APPROVED** [review]

File: srv/components/system/config/set_hostname.sls [hunk 2]
```diff
@@ -15,9 +15,12 @@
 # please email opensource@seagate.com or cortx-questions@seagate.com.
 #
 
+Update hostname file permissions:
```
Comment (andkononykhin) [inline srv/components/system/config/set_hostname.sls:18]: could you make that more specific, e.g. `Make hostname file modifiable`

File: srv/components/system/config/set_hostname.sls [hunk 3]
```diff
@@ -15,9 +15,12 @@
...
+  cmd.run:
+    - name: "chattr -i /etc/hostname"
```
Comment (andkononykhin) [inline srv/components/system/config/set_hostname.sls:20]: shouldn't we restore that protection once hostname is updated?

**Review by andkononykhin — COMMENTED** [review]
@SomwanshiAnjali Please check y suggestion. Thanks

File: srv/components/system/config/set_hostname.sls [hunk 4]
```diff
@@ -15,9 +15,12 @@
...
+
 Set hostname:
-  network.system:
-    - name: set_hostname
-    - hostname: {{ pillar['cluster'][grains['id']]['hostname'] }}
-    - apply_hostname: True
-    - retain_settings: True
+  cmd.run:
+    - name: "hostnamectl set-hostname {{ pillar['cluster'][grains['id']]['hostname'] }}"
```
Comment (ypise) [inline srv/components/system/config/set_hostname.sls:24]: Prefer converting this to a custom module.
Then handle the `chattr` driven immutability inside the custom set_hostname module.

**Review by ypise — CHANGES_REQUESTED** [review]

File: srv/components/system/config/set_hostname.sls (hunk 1 above)
Comment (SomwanshiAnjali) [reply srv/components/system/config/set_hostname.sls:19]: @83bhp 
yes we can use `network.system`, verified and updated PR accordingly

File: srv/components/system/config/set_hostname.sls (hunk 2 above)
Comment (SomwanshiAnjali) [reply srv/components/system/config/set_hostname.sls]: Updated

File: srv/components/system/config/set_hostname.sls (hunk 3 above)
Comment (SomwanshiAnjali) [reply srv/components/system/config/set_hostname.sls:20]: Updated

File: srv/components/system/config/set_hostname.sls (hunk 4 above)
Comment (SomwanshiAnjali) [reply srv/components/system/config/set_hostname.sls]: Used `network.system` module instead of set-hostname command

**Review by ypise — APPROVED** [review]
Agreeing to converting the logic to a custom module in separate PR.

**Pull request merged**